internal/provider/resource_k8s.go

```diff
@@ -464,52 +503,60 @@ func (k *K8sDeployer) storeState(d *schema.ResourceData, cl *apiClient) {
 	k.retainChecksums(workers, master)
 
 	l := []interface{}{master}
-	k.updateNetworkState(d, cl.state)
+	k.updateNetworkState(cl.substrateConn, d, cl.state)
 	d.Set("master", l)
 	d.Set("workers", workers)
 	d.Set("token", k.Token)
 	d.Set("ssh_key", k.SSHKey)
 	d.Set("network_name", k.NetworkName)
-	d.Set("node_deployment_id", nodeDeploymentID)
+	d.Set("capacity_deployment_map", capacityDeploymentMap)
 }
 
-func (k *K8sDeployer) updateNetworkState(d *schema.ResourceData, state state.StateI) {
+func (k *K8sDeployer) updateNetworkState(sub *substrate.Substrate, d *schema.ResourceData, state state.StateI) {
 	ns := state.GetNetworkState()
 	network := ns.GetNetwork(k.NetworkName)
-	before, _ := d.GetChange("node_deployment_id")
-	for node, deploymentID := range before.(map[string]interface{}) {
-		nodeID, err := strconv.Atoi(node)
+	before, _ := d.GetChange("capacity_deployment_map")
+	for capacityIdStr, deploymentID := range before.(map[string]interface{}) {
+		capacityId, err := strconv.ParseUint(capacityIdStr, 10, 64)
 		if err != nil {
 			log.Printf("error converting node id string to int: %+v", err)
 			continue
 		}
 		deploymentIDStr := fmt.Sprint(deploymentID.(int))
-		network.DeleteDeployment(uint32(nodeID), deploymentIDStr)
+		nodeId, err := getNodeIdByCapacityId(sub, capacityId)
+		// shouldn't panic if we failed to get the node to free the ip
+		if err != nil {
+			continue
+		}
+		network.DeleteDeployment(nodeId, deploymentIDStr)
```

Comment:
it's very weird that delete deployments takes `deploymentIDStr`

Reply:
will handle it after merge because the net state is used in different places, not in k8s only to not break others prs